webapp/channels/src/components/cloud_preview_modal/cloud_preview_modal.tsx

```diff
@@ -19,26 +24,33 @@ const CloudPreviewModal: React.FC = () => {
     const subscription = useSelector(getCloudSubscription);
     const license = useSelector(getLicense);
     const currentUserId = useSelector(getCurrentUserId);
+    const team = useSelector(getCurrentTeam);
 
     const isCloud = license?.Cloud === 'true';
     const isCloudPreview = subscription?.is_cloud_preview === true;
 
     // Check if modal has been shown before
-    const hasModalBeenShown = false;
+    const hasModalBeenShown = useSelector((state: GlobalState) => getPreference(state, CLOUD_PREVIEW_MODAL_SHOWN_PREF, CLOUD_PREVIEW_MODAL_SHOWN_PREF)) === 'true';
 
     const [showModal, setShowModal] = useState(false);
 
+    const filteredContentByUseCase = (content: PreviewModalContentData[]) => {
+        return content.filter((content) => content.useCase === team?.name.replace('-hq', ''));
```

Comment:
To make sure I understand how this feature works, am I right in understanding that we're automatically creating teams on preview environments based on the use case selected and then we're showing showing this modal when the user first visits the pregenerated team?

Reply:
That's correct. The data is pre seeded with 3 use cases, one of which is selected by the user on mattermost.com on entry to the flow. The seeded data has team names with suffix -hq (ie mission-ops-hq) so this filters that out to send them to the right place.